### DISCUSSION TITLE
如何向別人提問

### DISCUSSION THREAD

**Discussion by sunset513** [1101-GS4719-C/Python-Community-GS4719-C#1916] [優秀專案]
## 簡介

我感覺這篇不太該歸類在實用專案，他比較像是一本工具書指南，但他是個 repository 所以我想說就丟來這類好了www

這個 repository 主要是提供了一個指南，教導人們如何以聰明的方式提問問題，尤其是在技術社群中，這樣的溝通技巧非常重要!
這個指南涵蓋了許多方面，包括如何準備好提問、如何提供足夠的背景資訊、如何表達問題、以及如何有效地與他人溝通和互動等等。

## 專案連結
[[ryanhanwu / How-To-Ask-Questions-The-Smart-Way](https://github.com/ryanhanwu/How-To-Ask-Questions-The-Smart-Way)]

## 心得

對於許多新手來說，學習如何正確地提問問題可能會是一個挑戰，因為新手們可能不確定應該提供多少背景資訊，如何清晰地表達問題，以及如何尊重他人的時間和資源，避免引起其他人的不耐煩。

這個 repository 不僅僅提供了技術性的建議，還包括了一些溝通和互動方面的技巧，這對我這種剛開始進入技術領域的學生來說非常重要。我希望能夠和他人建立良好的關係，並且能夠得到有效的幫助和支持，但同時我又害怕別人會不耐煩回答我的問題。好在有這 repository 提供了我許多方向，就像我人生道路中的一盞明燈。

我覺得很重要的一段是: 絕不要自以為夠格得到答案，因為你沒有為這種服務支付任何報酬。我應該要自己探索到一個答案，透過提出有內涵的、有趣的、有思維激勵作用的問題，為這個社群貢獻，而不僅僅是被動的從他人處索取知識。

總的來說，這個 repository 對於學生來說是一個非常有價值的資源，它提供了實用的技巧和建議，幫助我們更好地進行溝通和互動。這對學習和成長都非常有幫助，讓我們能夠更快地掌握技術，並且更好地融入技術社群中。

**Comment by OG-Matcha** [comment]
同學的分享真的非常實用，我們都知道學習新東西前要先知道如何提問，去學習這種提問的藝術我覺得對每個人都非常實用，這種換位思考的概念值得大家去學習，非常感謝同學的分享讓大家知道。

**Comment by benbebobo149** [comment]
提問的藝術確實是需要練習的，並不是所有人一開始就會學會怎麼正確的發問。因此更重要的是保持一個良好的心態，不要因為別人的批評就不敢發問，要先想好說那些批評你問題很爛的人，它本身的發言有沒有建設性，如果沒有就不要理他。他只是在秀他的下限而已。對我來說，學會提出有建設性的問題，並且勇於尋求幫助和學習，比一開始就完美無缺更為重要。最後，我要感謝同學的分享，讓我有機會思考和成長。

**Comment by LinChloe** [comment]
給你這篇貼文給個讚!
從上大學後遇到許多問題，明明都是上網查詢，發現別人能解決的，但我卻不行。後來發現是提問方式有問題，需要像誰提問，用字遣詞等都是重點。像一個問題，你問google或Chatgpt，不一定誰會比較好。有時我問完問題都覺得我的表達方式很爛，不期望對方聽得懂。而這需要時間和經驗去累積，問多了就會慢慢清楚自己該如何發問。但在這之前，有一個厚臉皮和強硬的心臟，並且不恥下問才是重點。

**Comment by wanka2030** [comment]
這篇文章能夠讓一些不敢提出問題的學生能夠有更多的勇氣發問。對我來說，我不常去向他人提出問題，總是不斷地去嘗試、挑戰，才能夠得出正確的解法。假如能夠利用書中提到的上網搜尋、慢慢提問以及找出同溫層來獲得解法，就可以讓自己跨出第一步，不用再撞到南牆後，或者浪費更多的時間來做出毫無意義的事。能夠有可以回答問題的同學，更可以讓自己的學習更有效率:D
